packages/ui/src/components/language-selector/index.tsx

```diff
@@ -0,0 +1,130 @@
+import { useEffect, useState } from 'react'
+
+import { Button, ButtonGroup, Dialog } from '@/components'
+import { CheckIcon } from '@radix-ui/react-icons'
+
+export enum Language {
+  English = 'English',
+  Bulgarian = 'Bulgarian',
+  Croatian = 'Croatian',
+  Czech = 'Czech',
+  French = 'French',
+  German = 'German – Standard',
+  Irish = 'Irish – Extended',
+  Russian = 'Russian – QWERTY',
+  LatinAmerican = 'Latin American'
+}
+
+export enum LanguageCode {
+  EN = 'EN',
+  БГ = 'БГ',
+  HR = 'HR',
+  CZ = 'CZ',
+  FR = 'FR',
+  DE = 'DE',
+  IE = 'IE',
+  PY = 'PY',
+  LA = 'LA'
+}
+
+export interface LanguageInterface {
+  code: LanguageCode
+  name: Language
+}
+
+const languages: LanguageInterface[] = [
+  { code: LanguageCode.EN, name: Language.English },
+  { code: LanguageCode.БГ, name: Language.Bulgarian },
+  { code: LanguageCode.HR, name: Language.Croatian },
+  { code: LanguageCode.CZ, name: Language.Czech },
+  { code: LanguageCode.FR, name: Language.French },
+  { code: LanguageCode.DE, name: Language.German },
+  { code: LanguageCode.IE, name: Language.Irish },
+  { code: LanguageCode.PY, name: Language.Russian },
+  { code: LanguageCode.LA, name: Language.LatinAmerican }
+]
+
+interface LanguageDialogProps {
+  defaultLanguage?: LanguageCode
+  language?: LanguageCode
+  open: boolean
+  onOpenChange: (open: boolean) => void
+  onChange: (language: LanguageInterface) => void
+  onSave: (language: LanguageInterface) => void
+  onCancel: () => void
+  children: React.ReactNode
+}
+
+const LanguageDialog: React.FC<LanguageDialogProps> = ({
+  defaultLanguage,
+  language,
+  open,
+  onOpenChange,
+  onChange,
+  onSave,
+  onCancel,
+  children
+}) => {
+  const [selectedLanguage, setSelectedLanguage] = useState<LanguageCode | null>(null)
+
+  useEffect(() => {
+    if (language) {
+      setSelectedLanguage(language)
+    } else if (defaultLanguage) {
+      setSelectedLanguage(defaultLanguage)
+    }
+  }, [defaultLanguage, language])
```

Comment:
The language is controlled state both within the component and externally. This means that state will persist even if the user cancels out.

Reply:
Agreed, this is cleaned up in https://github.com/harness/canary/pull/921.